src/Internal/Gates.hs

```diff
@@ -47,21 +49,12 @@ runGate g x = do
     return x
 
 -- run specified gates in parallel
--- TODO
-parallel :: [Matrix C] -> QM ()
-parallel = undefined
-
--- | Projection of the zero basis vector
-proj0 :: Matrix C
-proj0 = (2 >< 2)
-  [ 1 , 0
-  , 0 , 0 ]
-
--- | Projection of the one basis vector
-proj1 :: Matrix C
-proj1 = (2 >< 2)
-  [ 0 , 0
-  , 0 , 1 ]
+-- TODO: should work, but not fully tested. does NOT work with controlled gates
+parallel :: Int -> [(Matrix C, QBit)] -> Matrix C
```

Comment:
great!

Reply:
Make sure to clearly define what happens when the list of gates is smaller than the number of qubits